src/OaiPmh/Harvester.php

```diff
@@ -340,10 +336,20 @@ protected function getRecordsByToken($token)
      */
     protected function storeDateSettings($settings)
     {
+        // Get identity information, including current OAI host time.
+        $response = $this->sendRequest('Identify');
+        $servertime = (string)$response->responseDate;
+        // Autoload granularity if necessary:
+        if ($this->granularity == 'auto') {
+            $this->granularity = (string)$response->Identify->granularity;
+        }
         // Set up start/end dates:
         $from = empty($settings['from'])
             ? $this->stateManager->loadDate() : $settings['from'];
-        $until = empty($settings['until']) ? null : $settings['until'];
+        // If no end-date is specified use the current server time. Tracking an
```

Comment:
As discussed in #7, I still think we're better off leaving things open-ended when no until is specified. What you're doing here makes perfect sense within the specification, but there are a lot of faulty implementations out there, so it's better to err on "too many records" than "not enough records."

Reply:
I agree, and I appreciate the discussion that led to this conclusion.

Reply:
https://github.com/vufind-org/vufindharvest/pull/8/commits/ccd0f59ff3581c302a917b2b4a1e93079b3fb990 now incorporates some changes directly to launch() that preserves the old null handling for "until". When this value is null, last_harvest.txt is sourced separately from the OAI server's clock. All the state preservation logic is still preserved. I think this would give us the best of all worlds as per the issue discussion.

Reply:
Yes, makes sense to me!